src/Workspaces/VisualBasic/Portable/Utilities/SpeculationAnalyzer.vb

```diff
@@ -140,6 +140,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Utilities
                 Case SyntaxKind.RangeArgument
                     semanticModel.TryGetSpeculativeSemanticModel(position, DirectCast(nodeToSpeculate, RangeArgumentSyntax), speculativeModel)
                     Return speculativeModel
+
+                Case SyntaxKind.AsNewClause
```

Comment:
Is there a way the check for what is supported can be integrated with this method so they don't get out of sync?